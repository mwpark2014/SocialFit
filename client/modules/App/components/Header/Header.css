.header {
  background-size: cover;
  border-bottom: 1px solid #ccc;
}

.content {
  width: 100%;
  margin: 0 15px;
  overflow: auto;
}

@media (min-width: 768px) {
  .nav {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

a.site-title.site-title{
  font-weight: 500;
  font-size: 34px;
  color: rgba(0, 0, 0, 0.6);
  padding-top:0px;
  padding-bottom:0px;
}

.loginButton {
  margin: 0 10px;
}

@media (min-width: 768px) {
  :global .navbar-expand-md .navbar-nav .nav-link {
    padding-right:2rem;
  }
}

@media (max-width: 767px){
  .add-post-button{
    float: left;
    margin-top: 16px;
  }
}

.clearfix::after {
  clear: both;
  display: block;
  content:"";
}