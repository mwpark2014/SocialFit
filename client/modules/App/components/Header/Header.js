import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import { Button } from 'react-bootstrap';

// Import Style
import styles from './Header.css';

export function Header(props, context) {
  console.log(props);
  let logButton = null;
  if (context.router.isActive('/', true)) {
    logButton = props.loginButton
    ? <Button href="#" className={styles.loginButton} onClick={props.deactivateLoginButton}> Sign in </Button>
    : <Button href="#" className={styles.loginButton} onClick={props.activateLoginButton}> Register </Button>;
  } else {
    logButton = props.authenticated
    ? <Button href="#" className={styles.loginButton} onclick={props.history.push('/')}> Log out </Button>
    : <Button href="#" className={styles.loginButton} onClick={props.history.push('/')}> Sign in </Button>;
  }

  return (
    <div className={styles.header}>
      <div className={styles.content}>
        <h1 className={styles['site-title']}>
          <Link to="/" ><h1>SocialFit</h1></Link>
        </h1>
        {logButton}
      </div>
    </div>
  );
}

Header.contextTypes = {
  router: React.PropTypes.object,
};

Header.propTypes = {
  authenticated: PropTypes.bool.isRequired,
  activateLoginButton: PropTypes.func.isRequired,
  deactivateLoginButton: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
  loginButton: PropTypes.bool.isRequired,
};

export default Header;
