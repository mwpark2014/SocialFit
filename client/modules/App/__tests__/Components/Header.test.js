import React from 'react';
import Header from '../../components/Header/Header';
import renderer from 'react-test-renderer';

const activateLoginButton = () => {
  this.props.dispatch(activateLoginButton());
};

const deactivateLoginButton = () => {
  this.props.dispatch(deactivateLoginButton());
};

const redirectToWelcomePage = () => {
  this.props.history.push('/');
};

test('Header snapshot', () => {
  const component = renderer.create(
    <Header
      authenticated={false} activateLoginButton={activateLoginButton}
      deactivateLoginButton={deactivateLoginButton} loginButton
      redirectToWelcomePage={redirectToWelcomePage} location={{ pathname: '/' }}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot(); // eslint-disable-line
});
