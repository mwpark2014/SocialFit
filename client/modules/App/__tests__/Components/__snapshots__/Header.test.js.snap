// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-faded"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        SocialFit
      </a>
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/feed"
              onClick={[Function]}
            >
              NEWSFEED
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              PROFILE
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="btn btn-secondary"
              href="#"
              onClick={[Function]}
              type={undefined}
            >
               Sign in 
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
